integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/clients/SimpleHttpClient.java

```diff
@@ -232,6 +235,41 @@ public HttpResponse doDelete(String url, final Map<String, String> headers) thro
     }
 
     /**
+     * Send a HTTP DELETE request with entity body to the specified URL
+     *
+     * @param url         Target endpoint URL
+     * @param headers     Any HTTP headers that should be added to the request
+     * @return Returned HTTP response
+     * @throws IOException If an error occurs while making the invocation
+     */
+    public HttpResponse doDeleteWithPayload(String url, final Map<String, String> headers,
+            final String payload, String contentType) throws IOException {
+
+        HttpUriRequest request = new HttpDeleteWithEntity(url);
+        setHeaders(headers, request);
+        HttpEntityEnclosingRequest entityEncReq = (HttpEntityEnclosingRequest) request;
+        final boolean zip = headers != null && "gzip".equals(headers.get(HttpHeaders.CONTENT_ENCODING));
```

Comment:
Thanks, Fixed in https://github.com/wso2/product-ei/pull/896/commits/80ac59e96e5c58a0751db3cf6ed056623f11b2bb